spec/requests/api/registrations_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe 'User Registration', type: :request do
+  let(:headers) { { HTTP_ACCEPT: 'application/json' } }
+
+  context 'with valid credentials' do
+    it 'returns a user and token' do
+      post '/api/auth', params: {email: 'me@mail.com',
+                                 password: 'whatever',
+                                 password_confirmation: 'whatever',
+                                 first_name: 'John',
+                                 last_name: 'Doe',
+                              }, headers: headers
+
+      expect(JSON.parse(response.body)['status']).to eq 'success'
+      expect(response.status).to eq 200
+    end
+  end
+
+  context 'returns an error message when user submits' do
+    it 'non-matching password confirmation' do
+      post '/api/auth', params: {email: 'me@mail.com',
+                                 password: 'whatever',
+                                 password_confirmation: 'Whatever',
+                                 first_name: 'John',
+                                 last_name: 'Doe',
+                              }, headers: headers
+
+      expect(JSON.parse(response.body)['errors']['password_confirmation']).to eq ["doesn't match Password"]
+      expect(response.status).to eq 422
+    end
+
+    it 'an invalid email address' do
+      post '/api/auth', params: {email: 'me@mail',
+                                 password: 'whatever',
+                                 password_confirmation: 'whatever',
+                                 first_name: 'John',
+                                 last_name: 'Doe',
+                              }, headers: headers
+
+      expect(JSON.parse(response.body)['errors']['email']).to eq ['is not an email']
+      expect(response.status).to eq 422
```

Comment:
I know that O has two expect statements in his examples, but I would like you to have one expect per it block. Okay?

Reply:
Updated as per my understanding. Please have a look